ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java

```diff
@@ -466,32 +469,31 @@ public Node visitQuerySpecification(final SqlBaseParser.QuerySpecificationContex
             ((Table) fromRel.getRelation()).getName().getSuffix() + " does not exist."
         );
       }
-      for (final Field field : fromDataSource.getSchema().fields()) {
-        final QualifiedNameReference qualifiedNameReference =
-            new QualifiedNameReference(location, QualifiedName
-                .of(fromDataSource.getName() + "." + field.name()));
-        final SingleColumn newSelectItem =
-            new SingleColumn(qualifiedNameReference, field.name());
-        selectItems.add(newSelectItem);
-      }
+
+      addFieldsFromDataSource(selectItems, fromDataSource, location,
```

Comment:
nice refactor! I was trying to do something like this but gave up - you'll have to rebase #2473 though 😂